src/setup_node_env/openssl_legacy_provider/openssl_legacy_provider_enabled.js

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module.exports = function () {
+  var nodeOptions = process.env.NODE_OPTIONS
+    ? process.env.NODE_OPTIONS.split(' ').filter(Boolean)
+    : [];
+  var execOptions = process.execArgv;
+
+  var cliOpenSSLLegacyProvider = checkOpenSSLLegacyProvider(execOptions);
+  var envOpenSSLLegacyProvider = checkOpenSSLLegacyProvider(nodeOptions);
+
+  if (typeof cliOpenSSLLegacyProvider === 'boolean') return cliOpenSSLLegacyProvider;
+  return Boolean(envOpenSSLLegacyProvider);
+};
+
+function checkOpenSSLLegacyProvider(options) {
+  var openSSLLegacyProvider = null;
+  options.forEach(function (option) {
+    if (option === '--openssl-legacy-provider') {
+      openSSLLegacyProvider = true;
+    }
+    if (option === '--no-openssl-legacy-provider') {
+      openSSLLegacyProvider = false;
+    }
+  });
+  return openSSLLegacyProvider;
+}
```

Comment:
As far as I understand, you can just feature-detect this instead, by looking at the available Ciphers (that should also make the testing much simpler), eg:

```suggestion
const crypto = require('crypto');

// The blowfish cipher is only available when node is running with the --openssl-legacy-provider flag
module.exports = crypto.getCiphers().includes('blowfish');

```

Reply:
You could also do `crypto.getHashes().includes('md4')` with a similar effect